sormas-ui/src/main/java/de/symeda/sormas/ui/vaccination/VaccinationController.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2021 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.ui.vaccination;
+
+import java.util.function.Consumer;
+
+import com.vaadin.ui.Window;
+
+import de.symeda.sormas.api.Disease;
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.immunization.ImmunizationReferenceDto;
+import de.symeda.sormas.api.infrastructure.district.DistrictReferenceDto;
+import de.symeda.sormas.api.infrastructure.region.RegionReferenceDto;
+import de.symeda.sormas.api.person.PersonReferenceDto;
+import de.symeda.sormas.api.user.UserRight;
+import de.symeda.sormas.api.utils.fieldaccess.UiFieldAccessCheckers;
+import de.symeda.sormas.api.vaccination.VaccinationDto;
+import de.symeda.sormas.ui.UserProvider;
+import de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent;
+import de.symeda.sormas.ui.utils.VaadinUiUtil;
+
+public class VaccinationController {
+
+	public void create(
+		ImmunizationReferenceDto immunization,
+		Disease disease,
+		UiFieldAccessCheckers fieldAccessCheckers,
+		Consumer<VaccinationDto> commitCallback) {
+		create(immunization, null, null, null, disease, fieldAccessCheckers, false, commitCallback);
+	}
+
+	public void create(
```

Comment:
Add a comment that explains how either the immunization or the other fields are optional.

OR: Make this method private and add another public create method for the other use case.